src/pip/_internal/vcs/__init__.py

```diff
@@ -219,6 +239,7 @@ def get_netloc_and_auth(self, netloc, scheme):
         return netloc, (None, None)
 
     def get_url_rev_and_auth(self, url):
+        # type: (str) -> Tuple[str, Optional[str], Tuple[Optional[str], Optional[str]]]  # noqa: E501
```

Comment:
I'd prefer we make a `AuthInfo` alias for `Tuple[Optional[str], Optional[str]]` and use that here, instead of `# noqa`.